Firestore/Source/API/FSTUserDataConverter.mm

```diff
@@ -515,6 +515,13 @@ - (nullable FSTFieldValue *)parseData:(id)input context:(FSTParseContext *)conte
   input = self.preConverter(input);
   if ([input isKindOfClass:[NSDictionary class]]) {
     return [self parseDictionary:(NSDictionary *)input context:context];
+
+  } else if ([input isKindOfClass:[FIRFieldValue class]]) {
+    // parseSentinelFieldValue may add an FSTFieldTransform, but nothing should be included in
+    // the actual parsed data, so we don't add this path to our fieldMask and we return nil.
+    [self parseSentinelFieldValue:(FIRFieldValue *)input context:context];
```

Comment:
This code is no longer calling `[context appendToFieldMaskWithFieldPath:*context.path]` which doesn't seem right.

If I `update(["a.b": FieldValue.serverTimestamp()])` shouldn't there be a field mask for "a.b"?

Reply:
That makes sense. Thanks for explaining.